File: append_scenes.py
import sys

game_path = 'game.js'
with open('town.js') as f: town = f.read()
with open('fishing.js') as f: fishing = f.read()
with open('cave.js') as f: cave = f.read()
with open('defense.js') as f: defense = f.read()

with open(game_path) as f: game = f.read()

marker = "/* ────────────────────────────────────────────────────────────────\n   §8  PHASER GAME CONFIGURATION & LAUNCH\n   ──────────────────────────────────────────────────────────────── */"

scenes = f"\n{town}\n{fishing}\n{cave}\n{defense}\n"

game = game.replace(marker, scenes + marker)

old_list = "scene: [BootScene, MenuScene, DevMenuScene, FarmScene, HouseScene],"
new_list = "scene: [BootScene, MenuScene, DevMenuScene, FarmScene, HouseScene, TownScene, FishingScene, CaveScene, DefenseScene, GameOverScene],"

game = game.replace(old_list, new_list)

with open(game_path, 'w') as f: f.write(game)
print("Done")

File: append_scenes.js
const fs = require('fs');

const gameJsPath = '/Users/studytime/Downloads/rpg game/game.js';
const townJs = fs.readFileSync('/Users/studytime/Downloads/rpg game/town.js', 'utf8');
const fishingJs = fs.readFileSync('/Users/studytime/Downloads/rpg game/fishing.js', 'utf8');
const caveJs = fs.readFileSync('/Users/studytime/Downloads/rpg game/cave.js', 'utf8');
const defenseJs = fs.readFileSync('/Users/studytime/Downloads/rpg game/defense.js', 'utf8');

let gameJs = fs.readFileSync(gameJsPath, 'utf8');

// Insert the new scenes right before the config section
const configMarker = "/* ────────────────────────────────────────────────────────────────\n   §8  PHASER GAME CONFIGURATION & LAUNCH\n   ──────────────────────────────────────────────────────────────── */";

const scenesBlock = `\n${townJs}\n${fishingJs}\n${caveJs}\n${defenseJs}\n`;

gameJs = gameJs.replace(configMarker, scenesBlock + configMarker);

// Update config scenes list
const oldSceneList = "scene: [BootScene, MenuScene, DevMenuScene, FarmScene, HouseScene],";
const newSceneList = "scene: [BootScene, MenuScene, DevMenuScene, FarmScene, HouseScene, TownScene, FishingScene, CaveScene, DefenseScene, GameOverScene],";

gameJs = gameJs.replace(oldSceneList, newSceneList);

fs.writeFileSync(gameJsPath, gameJs);
console.log("Appended scenes and updated config in game.js!");
